jenkins/image_building/build-image.sh

```diff
@@ -26,21 +26,20 @@ source "${current_dir}/upload-ci-image.sh"
 sudo sed -i "s/^#\$nrconf{restart} = 'i';/\$nrconf{restart} = 'a';/" /etc/needrestart/needrestart.conf > /dev/null || true
 
 sudo apt-get update
-sudo apt-get install python3-pip qemu qemu-kvm -y
-sudo pip3 install diskimage-builder python-openstackclient
-sudo -H pip install virtualenv
+sudo apt-get install python3-dev python3-pip python3-venv qemu qemu-kvm -y
+python3 -m venv venv
+
+# shellcheck source=/dev/null
+source venv/bin/activate
```

Comment:
```suggestion
. venv/bin/activate
```
source is deprecated